src/main/java/com/javadiscord/jdi/internal/api/DiscordRequestBuilder.java

```diff
@@ -0,0 +1,84 @@
+package com.javadiscord.jdi.internal.api;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DiscordRequestBuilder {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private String method;
+    private String url;
+    private String body;
+    private DiscordResponseFuture future;
```

Comment:
pre initialize this?

Reply:
Let's see how it pans out and leave this as future tech debt

Reply:
no, this will ensure user can take the future and pass it around instead of passing the entire request object

Reply:
I'm a bit confused, do you mean pre-init the fields or the `DiscordResponseFuture`

Reply:
done!